src/common/snippets/docs/mha_optimization_guide.md

```diff
@@ -0,0 +1,81 @@
+- `SplitDimensionM::can_be_optimized` is used in CPU plugin callback: if this method reports that appropriate parallel work amount can not be set for the MHA, the tokenization doesn't happen.
+
+## Brgemm Blocking
+
+Within the Snippets CPU backend, the MatMul is executed using the Brgemm primitive. For enhancing the execution efficiency, blocking across the M, K, and N matmul dimensions is used.
+
+### Blocking Parameters
+
+The heuristics for determining the optimal block sizes can be found in [SetBrgemmCPUBlockingParams](../../../plugins/intel_cpu/src/transformations/snippets/x64/pass/set_brgemm_cpu_blocking_params.cpp).
+
+**Important notes:**
+- Blocking by M dimension is shared between both Brgemms. Please see [SplitLoops](../include/snippets/lowered/pass/split_loops.hpp) lowered pass for the details.
+- The heuristics from `SetBrgemmCPUBlockingParams` are not currently applied by K and N dimensions for BF16/INT8 Brgemms, although it is technically supported.
+
+### Blocking Order
+
+The lowered pass [BrgemmBlocking](../../../plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/brgemm_blocking.cpp) performs blocking loops creation on LinearIR. Currently, the order of blocking loops is following (from outer to inner): `M->N->K`.
+
+## MHA Performance Tuning Recommendations
+
+Buiding on the information presented upper, this section focuses on providing recommendations for fine-tuning the performance of specific MHA pattern.
+
+1. Check if there are MHA's which were not tokenized because of [CPU plugin callback](#cpu-plugin-callback-for-mha-tokenization).
+2. Check how the graph was changed by [CommonOptimizations](#snippets-common-optimizations). In local experiments, some transformations might be worth to change:
+    - Disable [ExtractUnsupportedTransposes](#extractunsupportedtransposes) transformation in order to benchmark Snippets Transpose implementation.
+    - Adjust [SplitDimensionM](#splitdimensionm) heuristics in order to benchmark another splitting, or disable the pass at all.
+3. [Blocking parameters](#blocking-parameters): adjust blocking heuristics in `SetBrgemmCPUBlockingParams`.
+    - Please note that there is 2 Matmul nodes inside a single MHA, and each Matmul can have his own optimal K, N blocking params. M block is better to keep the same since the corresponding blocking loop is shared between both Matmuls.
+    - For the BF16/INT8 blocking loops, 2 options are possible: blocking can be done only for Brgemm node, or for BrgemmCopyB repacking too.
+4. [Blocking order](#blocking-order): blocking order changing might have an effect as well. But please note, that blocking by M dimensions is shared between both Brgemms only if blocking by M is outermost.
+
+By following these recommendations, signifficant performance improvement for the specific MHA can be achieved. Additionally, the results of these experiments can be used as a solid foundation for the subsequent heuristics adjusting.
```

Comment:
> signifficant performance improvement for the specific MHA can be achieved.
After reading this sentence, one might think that we don't provide adequate out-of-the box performance for most of the patterns 🙂 
Let's paraphrase a little bit:
```suggestion
Following these recommendations, the performance of some specific MHA patters can be fine-tuned. Additionally, the results of these experiments can be used as a solid foundation for the subsequent heuristics adjustments. 
```